core/lib/workarea/release.rake

```diff
@@ -0,0 +1,105 @@
+require 'active_support/inflector'
+
+load 'workarea/changelog.rake'
+
+namespace :workarea do
+  desc "Release the next version of this gem"
+  task :release, [:private] do |task, args|
+    #
+    # Devise current gem version
+    #
+    #
+    version_file_path = Dir['**/version.rb'].reject { |p| p =~ /vendor/ }.first
+    gem_namespace =
+      version_file_path
+        .split('lib/').last
+        .split('/version').first
+        .classify
+
+    gem_name = gem_namespace.gsub(/::/, ' ')
+
+    current_version = if gem_namespace == 'Workarea'
+      "#{gem_namespace}::VERSION::STRING".constantize
+    else
+      "#{gem_namespace}::VERSION".constantize
+    end
+
+    #
+    # Devise next gem version
+    #
+    #
+    current_branch = `git rev-parse --abbrev-ref HEAD`.chomp
+    puts <<~MESSAGE
+      You're about to release a new version of the #{gem_name} gem.
+
+      The current version in the #{current_branch} branch is #{current_version}.
+
+      Please provide the version that you'd like to release:
+    MESSAGE
+
+    next_version = STDIN.gets.chomp
+
+    puts "Should I release #{next_version} from #{current_branch}? (Y/n)"
+    answer = STDIN.gets.chomp
+    answer = 'Y' if answer.empty?
+    abort("Nothing to do") unless answer.casecmp?('Y')
+
+    #
+    # Update version.rb and CHANGELOG.md
+    #
+    #
+    tempfile = Tempfile.new('version_temp.rb')
+    File.open(version_file_path, 'r') do |file|
+      file.each_line do |line|
+        tempfile.puts line.include?(current_version) \
+          ? line.gsub(current_version, next_version) \
```

Comment:
Is this reliable? Feels... I dunno. Unreliable or something.

Reply:
This is pretty reliable. Now with the change to base's `version.rb` files all gems have a consistent string that will be replaced during the update.

Reply:
instead of parsing the `version.rb` file, we should just be reading the `Workarea::GemName::VERSION` constant that should already be loaded in Rakefile.